### PR TITLE
Add warning about implicit dependency on react-addons-test-utils

### PR DESCRIPTION
**Pull request by thatjessicakelly**
A warning like this should hopefully help with #225.

Not exactly sure what the text should say.

### CODE REVIEW

**Comment by lelandrichardson** [comment]
Hmm. This is a good idea. we should do this for all of the implicit dependencies. We could link to the README of the repo for more context as well

File: src/react-compat.js [hunk 1]
```diff
@@ -70,7 +70,16 @@ if (REACT013) {
...
-  TestUtils = require('react-addons-test-utils');
+  try {
+    TestUtils = require('react-addons-test-utils');
+  } catch (e) {
+    console.error(
+      'react-addons-test-utils is an implicit dependency in order to support react@0.13-14. ' +
+      'Please add the appropriate version to your devDependencies. ' +
+      'See https://github.com/airbnb/enzyme/issues/225'
+    );
+    throw e;
```
Comment (ljharb) [inline src/react-compat.js]: tbh this should probably `process.exit(1)` rather than rethrowing the error

**Comment by ljharb** [comment]
How many other implicit non-peer deps are there?? ideally we have as close to zero as possible

File: src/react-compat.js (hunk 1 above)
Comment (lelandrichardson) [inline src/react-compat.js]: I think error is better. Enzyme can be run in browser-based test runners

Comment (ljharb) [inline src/react-compat.js]: ah, very true, this is fine then

**Comment by lelandrichardson** [comment]
We are in this situation because of 0.13 and 0.14 compatibility...

If you are using React 0.13, then you are all good. `react` is all you need, and react is a dep.
If you are using React 0.14, then you need to have `react`, `react-dom`, and `react-addons-test-utils`.

Since `react-dom` and `react-addons-test-utils` are packages that cannot be installed alongside react 0.13, we need to have implicit dependencies.

**Comment by ljharb** [comment]
I wonder if making them `optionalDependencies` would be any better - i'm not sure how that field interacts with npm

**Comment by lelandrichardson** [comment]
@ljharb optionalDependencies won't work. We tried that first, and shit blew up if you tried to use enzyme with 0.13

**Comment by ljharb** [comment]
ah right, i'd forgotten. in that case this try/require/catch pattern is probably the cleanest way forward - maybe we could put all of those in one file (either one group or one per weird-dep) that re-exports either the libs or `undefined`, and require *that* module where they're needed?

**Comment by thatjessicakelly** [comment]
Let me know how you want me to handle it and I'm happy to update this

**Comment by lelandrichardson** [comment]
@ljharb i feel like there is already enough misdirection with the `react-compat` file. This file is already dealing with the differences between 0.13 and 0.14, so putting this logic here I think is sane.

@thatjessicakelly, do you think you could just update your PR to do the same thing for `react-dom` in addition to `react-addons-test-utils`?

**Comment by ljharb** [comment]
Makes sense.

**Comment by thatjessicakelly** [comment]
Done. I also linked to the README

**Comment by capaj** [comment]
@thatjessicakelly you keep using that word "warning", yet you throw an error. Interesting.

**Comment by ljharb** [comment]
@capaj that tone is not appreciated. Conceptually it's a warning, so by a technical definition of English, it's correct. You will not do well playing the condescending pedantry game here.

**Comment by capaj** [comment]
> condescending pedantry game here
ok :grinning:

**Pull request merged**